api/envoy/config/cluster/v3/cluster.proto

```diff
@@ -481,6 +481,11 @@ message Cluster {
       option (udpa.annotations.versioning).previous_message_type =
           "envoy.api.v2.Cluster.CommonLbConfig.ConsistentHashingLbConfig";
 
+      enum LbPolicy {
```

Comment:
Starting out with some API comments.

1) We should implement an entirely new load balancer for this as an extension. This will mean actually implementing this enum type which is not yet implemented: https://github.com/envoyproxy/envoy/blob/f6679d51cc7b2f0b5e05c883a035ad87d011f454/api/envoy/config/cluster/v3/cluster.proto#L112-L117 (This will mean also resolving https://github.com/envoyproxy/envoy/issues/5598)

2) Once we have a new extension LB for shuffle sharding, we can have an independent configuration for that load balancer. To implement I recommend sharing the maglev code as a base class in both load balances, but not substantially altering the existing maglev/ring hash code.